cdap-client/src/main/java/co/cask/cdap/client/AuthorizationClient.java

```diff
@@ -88,6 +88,11 @@ public void enforce(EntityId entity, Principal principal, Set<Action> actions) t
   }
 
   @Override
+  public Set<? extends EntityId> isVisible(Set<? extends EntityId> entityIds, Principal principal) throws Exception {
+    throw new UnsupportedOperationException("Visibility check is not supported via Java Client.");
```

Comment:
Is there any specific reason to disallow this? What will happen if a user build his own UI using our clients?